src/projectManager.ts

```diff
@@ -40,6 +52,11 @@ export class ProjectManager {
     }
 
     public static initialize(context: ExtensionContext) {
+        this.emitter.event(e => {
```

Comment:
What does this do?

Reply:
When an event is fired (`projects` or `activeProject`), we are filtering all the callback functions to get only the ones registered for the event that was fired and then calling them.

Reply:
Very cool, so why are we doing that on initialize. (I am asking for my education only)

Reply:
This `ProjectManager.initialize()` function is only called when the extension is first activated in order to do some setup. We do this here to just setup the `EventEmitter` to respond to events as described above.